cluster/node.go

```diff
@@ -0,0 +1,72 @@
+package cluster
+
+import (
+	"sync"
+	"time"
+)
+
+type NodeState int
+
+const (
+	NodeReady       NodeState = 0
+	NodeNotReady    NodeState = 1
+	NodeUnreachable NodeState = 2
```

Comment:
I have more questions about how that would work, but i will reserve those questions for the clustering PR. in the meantime I would suggest removing the field for now, and re-adding it in the cluster PR.